src/main.rs

```diff
@@ -10,18 +10,66 @@ extern crate pal_pc as pal;
 #[macro_use]
 extern crate log;
 
+#[macro_use]
+extern crate alloc;
+
+use client::FobnailClient;
 use smoltcp::iface::{EthernetInterfaceBuilder, Neighbor, NeighborCache};
 use smoltcp::socket::{SocketSet, UdpPacketMetadata, UdpSocket, UdpSocketBuffer};
 use smoltcp::time::Instant;
 use smoltcp::wire::{IpAddress, IpCidr, Ipv4Address};
 
+use coap::CoapClient;
+
+mod client;
+mod coap;
+
+// TODO: should check how much we actually need here and decrease (or increase)
+// these if needed
+const UDP_META_DEFAULT_BUF_LEN: usize = 16;
+const ECHO_SERVER_BUF_LEN: usize = 512;
+// Set at Ethernet MTU
+// Actually we can transfer less than that in UDP packet since Ethernet frame +
+// IP frame already takes some space. Should compute exatly how much bytes do we
+// need.
+const COAP_CLIENT_BUF_LEN: usize = 1500;
+
+// FIXME: should not use hardcoded server addresses, instead server should be
+// discovered
```

Comment:
Actually, Fobnail should make host use this (or any other in this network) address so it won't have to be manually assigned by user. Maybe hardcoded response to DHCP query from host?

Reply:
Right, by hardcoded I meant just IP, without having to implement a database, assignment algorithm etc. on DHCP server's side.

Ack on the updated comment.